package.json

```diff
@@ -191,6 +190,7 @@
     "chrome-launcher": "^0.15.1",
     "configstore": "^5.0.1",
     "csp_evaluator": "1.1.1",
+    "devtools-protocol": "0.0.1130274",
```

Comment:
This needs to be a package dep for type checking. This shouldn't make a big difference for the bundle size, since  `puppeteer-core` ships with this package and we synchronize the version.

Reply:
might that also cause problems if the `devtools-protocol` types ever conflict?

Reply:
That shouldn't happen because we have the resolution field. 

Digging in a little deeper, I think part of the problem in this PR also came from how `node_modules` is structured for different lock files...

Without this change, `lighthouse` could be pulling the `devtools-protocol` from sub-dependency `puppeteer-core`, or it could be pulling the `devtools-protocol` from the sibling dependency `puppeteer`. There is no gurantee, and this was part of the problem that we were getting in our type checking docs test.

With this change, `lighthouse` is guranteed to be pulling the same `devtools-protocol` version as `puppeteer-core`.

Reply:
I believe `resolutions` only works for ourselves, not if `lighthouse` is a dependency (and not at all if you're using npm). That was fine for when we were just type checking internally, but we may need a different approach now that we're being type checked as a dep (e.g. `devtools-protocol: ">=version-puppeteer-core uses"` since we use `^` with `puppeteer-core` and we're unlikely to convince them to be looser with their deps)